src/main/src/main/java/org/geoserver/security/ResourceAccessManager.java

```diff
@@ -71,4 +71,7 @@ public default LayerGroupAccessLimits getAccessLimits(
      */
     public @Nullable Filter getSecurityFilter(
             Authentication user, final Class<? extends CatalogInfo> clazz);
+
+    /** Checks if {@code user} has admin privileges on at least one workspace */
+    public boolean isWorkspaceAdmin(Authentication user);
```

Comment:
Can  you please make this a default method instead? 

In my work experience, ResourceAccessManager is one of the most commonly implemented extension points, typically to leverage installation specific rule management and/or to integrate with existing systems.
Implementations typically have no reason to implement the Abtract class, as it's mostly empty methods, and has been used only in tests.

Reply:
That was my first instinct as well, but then I thought it'd be weird that the method received the Catalog as parameter.
It'd have to be 
```java
default boolean isWorkspaceAdmin(Authentication user, Catalog catalog){...}
```
where no other method in ResourceAccessManager requires a catalog as argument.
If that's still good enough I can go for it, but had to mention it just in case you didn't notice

Reply:
I see what you mean. On the other side, DefaultResourceAccessManager haven't had a reference to catalog until 2016, when container layer group mode was introduced (a group mode that is not used often). The typical implementation of ResourceAccessManager out there won't necessarily have access to a catalog... checking ones I have handy, none of them do... as site-specific implementations, they don't care about group nesting to start with.

Anyhow, personally I'd be ok with the current implementation as well, as long as you [update the upgrade instructions](https://docs.geoserver.org/latest/en/user/installation/upgrade.html#notes-on-upgrading-specific-versions) mentioning the API breakage for existing ResourceAccessManager implementations.

Reply:
with that in mind, I actually prefer the default method on the interface. Honestly I thought you were going to oppose to that lol, as that was my first approach

Reply:
Okay, moved the method as a default one on the interface, and removed the need to extend AbstractResourceAccessManager altogether.